### ISSUE TITLE
自构建数据集loss出现NaN的问题

### ISSUE THREAD

**Issue by Laki-Cat** [WenmuZhou/PytorchOCR#300]
我使用的是自定义数据集，数据是通过将单字数据合成为文本行的方式构建的。在训练过程中，模型的 loss 出现了 NaN 的情况。而在此之前，我使用 TTF 字体文件合成了大量数据进行训练，并没有出现类似的问题。
下面的图片选自我合成的数据集

![Image](https://github.com/user-attachments/assets/dd4cc0d2-9af8-4dcd-82f6-f4d6c952af99)

![Image](https://github.com/user-attachments/assets/2ca6c6d8-4108-4276-b9bc-58d3e6ab09a1)

![Image](https://github.com/user-attachments/assets/9b4c3f54-d0a1-4910-977c-f576b1aa8a73)

![Image](https://github.com/user-attachments/assets/34b17f32-56e4-4903-9529-4255747d6d24)

**Comment by Laki-Cat** [comment]
Global:
  device: gpu
  epoch_num: 200
  log_smooth_window: 20
  print_batch_step: 10
  output_dir: ./ch_PP-OCRv4_rec_train/student.pth
  eval_epoch_step: [0, 1]
  cal_metric_during_train: true
  pretrained_model: 
  checkpoints:
  use_tensorboard: true
  infer_mode: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: &character_dict_path torchocr/utils/ppocr_keys_v1.txt
  max_text_length: &max_text_length 54
  use_space_char: &use_space_char true

Export:
  export_dir:
  export_shape: [ 1, 3, 48, 1000 ]
  dynamic_axes: [ 0, 2, 3 ]



Optimizer:
  name: Adam
  lr: 0.001
  weight_decay: 3.0e-05

LRScheduler:
  name: CosineAnnealingLR
  warmup_epoch: 5

Architecture:
  model_type: rec
  algorithm: SVTR_HGNet
  Transform:
  Backbone:
    name: PPHGNet_small
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode
  character_dict_path: *character_dict_path
  use_space_char: *use_space_char

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: ./PytorchOCR/datasets/val.txt
    ext_op_transform_idx: 1
    label_file_list:
    - /root/lanyun-tmp/PytorchOCR/datasets/datasets.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecConAug:
        prob: 0.5
        ext_data_num: 2
        image_shape: [48, 1000, 3]
        max_text_length: *max_text_length
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[1000, 32], [1000, 48], [1000, 64]]
    first_bs: &bs 89
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:
    shuffle: true
    batch_size_per_card: *bs
    drop_last: true
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /root/lanyun-tmp/PytorchOCR/datasets/
    label_file_list:
    - ./PytorchOCR/datasets/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 1400]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 100
    num_workers: 4


这是我的配置文件
